### ISSUE TITLE
Don't set hostname to service name when using --x-networking

### ISSUE THREAD

**Issue by vieux** [docker/compose#2327]
Using the new networking features, when you start a container.
The name is correctly `<folder>_<service>_1` but the hostname is `<service>`

It's impossible the scale the containers in some case because all the containers get the same hostname.

**Comment by vieux** [comment]
for example with:

```
$ cat docker-compose.yml
redis:
 container_name: redis
 image: redis
 command: redis-server

rqworker:
 image: vieux/rqworker
 command: -u redis://redis:6379/

rqdashboard:
 image: vieux/rq-dashboard
 command: -H redis
 ports:
  - 9181:9181
```

you cannot scale rqworker because the all end up with the hostname `rqworker` and it will fail.

**Comment by dnephin** [comment]
This is because of #2276. networking was supposed to use the hostname, but it didn't end up happening.

Can you tell me more about how it breaks? Isn't the hostname only internal to the container?

**Comment by vieux** [comment]
@dnephin it's an issue for me because `rqworker` advertises itself using it's hostname and rq refuses to have 2 workers with the same name.

```
Traceback (most recent call last):
  File "/usr/local/bin/rqworker", line 9, in <module>
    load_entry_point('rq==0.5.6', 'console_scripts', 'rqworker')()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/rq/cli/cli.py", line 202, in worker
    w.work(burst=burst)
  File "/usr/local/lib/python2.7/dist-packages/rq/worker.py", line 404, in work
    self.register_birth()
  File "/usr/local/lib/python2.7/dist-packages/rq/worker.py", line 238, in register_birth
    raise ValueError(msg.format(self.name))
ValueError: There exists an active worker named u'rqworker.1' already
```

**Comment by vieux** [comment]
@dnephin is there a way to hack the compose file to use

```
hostname: rqworker_{$COUNT}
```

**Comment by dnephin** [comment]
There is not. We'll have to fix this for 1.5.1
